src/megachatapi_impl.cpp

```diff
@@ -1081,8 +1081,10 @@ void MegaChatApiImpl::sendPendingRequests()
         case MegaChatRequest::TYPE_GET_FIRSTNAME:
         {
             MegaChatHandle uh = request->getUserHandle();
+            const char* publicHandle = request->getLink();
+            MegaChatHandle ph = publicHandle ? karere::Id(publicHandle).val : MEGACHAT_INVALID_HANDLE;
```

Comment:
The `Id` ctor should receive the length of the handle as a param (6 bytes). It probably won't make a difference, but it protects against uninitialized bytes in the higher part of the handle.